docs/development/devguide/tutorial/04-coding/08-persistence-data.md

```diff
@@ -0,0 +1,230 @@
+
+You can find the example files for the data class and the repository here:
+* [ilTodoItem](TodoExample/classes/class.ilTodoItem.php)
+* [ilTodoRepository](TodoExample/classes/class.ilTodoRepository.php)
+
+### Reading Data
+
+If you want to display the entries of a to-do list on the dashboard, you must read them from the database table. The following example generates a database query, executes it and creates an array of data objects that can then be processed further.
+
+````php
+    public function getItemsOfUser(int $user_id) : array
+    {
+        $items = [];
+        
+        $query = "SELECT * FROM todo_items WHERE user_id = " 
+            . $this->db->quote($user_id, 'integer');
+        
+        $result = $this->db->query($query);
+        while ($row = $this->db->fetchAssoc($result)) {
+            $items[] = new TodoItem(
+                $row['item_id'],
+                $row['user_id'],
+                $row['title'],
+                $row['description'],
+                $row['deadline']]
+            );
+        }
+        return $items;
+    }
+````
+
+1. the database query is defined as `$query`. It is important to insert all parameters of the query via the `quote()` function. In addition to the value to be inserted, it also receives its data type and inserts an appropriately formatted character string.
+2. the data is queried with `query()` and a result object is created.
+3. with `fetchAssoc()` the individual data records are read from the result object one after the other as associative arrays.
+
+As an alternative to `query()`, you can use the `queryF()` function, in which the query is passed with placeholders `%s` and the parameters are passed as arrays of data types and values. This can be clearer with many parameters, but makes it more difficult to output the specific query before it is executed when debugging. The example in [ilTodoRepository](TodoExample/classes/class.ilTodoRepository.php) is written this way.
+
+### Saving Data
+
+New entries on the to-do list must be saved in the table. The following example takes an entry as a data object and saves it as a new data record:
+
+````php
+    public function createItem(TodoItem $item) : TodoItem
+    {
+        $todo_id = $this->db->nextId('todo_items');
+        
+        $this->db->insert('todo_items', [
+            'todo_id' => ['integer', $todo_id],
+            'user_id' => ['integer', $item->getUserId()],
+            'title' => ['text', $item->getTitle()],
+            'description' => ['clob', $item->getDescription()],
+            'deadline' => ['date', $item->getDeadline()]
+        ]);
+        
+        return $item->widthTodoId($todo_id);
+    }
+````
+
+1. a new sequence value for the primary key field todo_id is generated with `nextId()`.
```

Comment:
maybe you could explain that `nextId` is only possible if the tables has been created with `createSequence`